### DISCUSSION TITLE
Passer le site en statique

### DISCUSSION THREAD

**Discussion by maxgfr** [SocialGouv/code-du-travail-numerique#5536] [Technique]
Récemment, j'ai merger une PR qui nous a permis de récupérer des `secret` au niveau du build de next.

Ainsi, cela nous permet de faire des calls depuis les `getStaticProps` sur `elasticsearch` durant la phase de build.

Le lien de la PR est ici : https://github.com/SocialGouv/code-du-travail-numerique/pull/5514

Par la suite, j'ai migré quelques pages qui étaient en `getServerSideProps` en `getStaticProps` afin de pouvoir générer au build time les pages.

C'est de l'[ISR](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration), j'ai mis une limite de 30 minutes, signifiant que chaque 30 minutes, la page se rebuild.

Le lien de la PR est là : https://github.com/SocialGouv/code-du-travail-numerique/pull/5515

Serait-ce worth-it de passer tout le site en statique ?

J'ouvre le débat ici.

**Comment by carolineBda** [comment]
ça serait top de passer nos pages informations qui ont des problème de perfs sur la Google Search Console. Et peut être aussi les pages contributions qui sont nos pages les plus importantes. pour le reste du site l’intérêt est un peu moindre

**Comment by carolineBda** [comment]
c'était déjà l'idée il y a 4 ans : https://github.com/SocialGouv/code-du-travail-numerique/issues/2606

**Comment by maxgfr** [comment]
La difficulté réside dans le load des scripts externe (matomo / react-dsfr). J'y réponds dans cette issue : #6822

Le `nonce` pour les CSP est passé via le `headers()`. L'autre possibilité est de passer un integrity hash au niveau du script afin d'avoir des CSP convenable ou alors utiliser une politique de la sorte : `script-src 'self' 'unsafe-inline';`
